doc: Avoid title case and make FAQs all questions (#11403)
ocaml/dune: doc/faq.rst

its ``public_name`` then that can be used interchangeably with the ``public_name``
in this command.

> Why are files and directories whose names begin with "." (period) are ignored by ``source_tree``?

Reviewer: We already have "are" at the start.

```suggestion
Why are files and directories whose names begin with "." (period) ignored by ``source_tree``?
```

Tho maybe something like this formulation would be a bit easier to parse out?

```suggestion
Why does ``source_tree`` ignore files and directories when they begin with a "." (period)?
```

    (dirs :standard .cargo \ target)
    (data_only_dirs vendor)

> How can I write inline tests in a package without my users needing to install ``ppx_inline_test``?

Reviewer: Because there is apparently no answer to "how", I think we can just answer the why not?

```suggestion
Why can't I write inline tests in a package without users needing to install ``ppx_inline_test``?
```
